arangodb-foxx-services/src/main/model.ts

```diff
@@ -15,4 +15,57 @@
  */
 
 export type DocumentKey = string
-export type ExecutionEvent = object
+export type ExecutionEvent = ExecutionEventLike
+export type DataSource = DataSourceLike
+
+interface DataSourceLike {
+    _key: string
+}
+
+interface ExecutionEventExtra {
+    appId: string
+}
+
+export interface ExecutionEventLike {
+    timestamp: bigint
+    extra: ExecutionEventExtra
+}
+
+/**
+ * mimics backend's [[za.co.absa.spline.consumer.service.model.LineageOverview]]
+ */
+export class LineageOverview {
+    readonly info: LineageOverviewInfo
+    readonly graph: LineageOverviewGraph
+
+    constructor(info: LineageOverviewInfo, graph: LineageOverviewGraph) {
+        this.info = info
+        this.graph = graph
+    }
+}
+
+export class LineageOverviewInfo {
+    readonly timestamp: bigint
+    readonly applicationId: string
+    readonly targetDataSourceId: string
+
+    constructor(timestamp: bigint, applicationId: string, targetDataSourceId: string) {
+        this.timestamp = timestamp
+        this.applicationId = applicationId
+        this.targetDataSourceId = targetDataSourceId
+    }
+}
+
+export class LineageOverviewGraph {
+    readonly depthRequested: number
+    readonly depthComputed: number
+    readonly nodes: Array<object>
+    readonly edges: Array<object>
+
+    constructor(depthRequested: number, depthComputed: number, nodes: Array<object>, edges: Array<object>) {
+        this.depthRequested = depthRequested
+        this.depthComputed = depthComputed
+        this.nodes = nodes
+        this.edges = edges
+    }
+}
```

Comment:
minor suggestion: 
You can write the same definition with less lines of code.
```suggestion
export class LineageOverviewGraph {
    constructor(
        readonly depthRequested: number,
        readonly depthComputed: number,
        readonly nodes: Array<object>,
        readonly edges: Array<object>) {
    }
}
```

Reply:
In general, why do you use classes/interfaces for merely defining complex data types? Classes and interfaces are OO abstractions and are useful when you add behavior (especially interfaces are there to describe public contract primarily in terms of methods). In this case, I would rather use just `type`

Reply:
The suggested change - amazing, I didn't know this - I tried to learn TS basics from https://www.typescriptlang.org/docs/handbook/2/classes.html, but that was not apparent.

I use the classes/interfaces because the object I access fields on, e.g. `executionEvent.timestamp` -- and the linter/IDE warns me about that field not being present - so I define a (at least partially) the class/interface to have the fields as part of the contract.

```typescript
export type ExecutionEvent = object
executionEvent.timestamp
```
gets you a warning (not expecting for `object.timestamp` to be available/correct), while
```typescript
export type ExecutionEvent = ExecutionEventLike
export interface ExecutionEventLike {
    timestamp: bigint
    ...
}
executionEvent.timestamp
```
seems to be ok (the contract is defined by the interface).

Or maybe I am missing something and you mean to use `type` differently?

Reply:
Suggestion applied in all similar constuctors.

Reply:
Thanks for the tip, I have used types as suggested in the end.